### PR TITLE
Fix the camera and selection bugs

### PR DESCRIPTION
**Pull request by eggrobin**
Today in "KSP is dumb": the FloatingOrigin system puts the origin where the root part of the vessel is (i.e. at top typically). Then the camera is offset from that so that it's at the barycentre of the vessel. Because putting the barycentre of the thing to which physics happens at 0 and the camera looking there would be too sane.

### CODE REVIEW

File: ksp_plugin/plugin.cpp [hunk 1]
```diff
@@ -624,34 +613,36 @@ void Plugin::AdvanceTime(Instant const& t, Angle const& planetarium_rotation) {
 }
 
 DegreesOfFreedom<World> Plugin::GetPartActualDegreesOfFreedom(
-    PartId const part_id) const {
-  CHECK(bubble_barycentre_);
+    PartId const part_id,
+    PartId const part_at_origin) const {
+  auto const world_origin = FindOrDie(part_id_to_vessel_, part_at_origin)
+                                ->part(part_at_origin)
+                                ->degrees_of_freedom();
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: We don't like arrows at the beginning of lines.

File: ksp_plugin/plugin.cpp [hunk 2]
```diff
@@ -624,34 +613,36 @@ void Plugin::AdvanceTime(Instant const& t, Angle const& planetarium_rotation) {
...
-  return *bubble_barycentre_;
-}
-
 DegreesOfFreedom<World> Plugin::CelestialWorldDegreesOfFreedom(
-    Index const index) const {
+    Index const index,
+    PartId const part_at_origin) const {
+  auto const world_origin = FindOrDie(part_id_to_vessel_, part_at_origin)
+                                ->part(part_at_origin)
+                                ->degrees_of_freedom();
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Same.

File: ksp_plugin/plugin.hpp [hunk 3]
```diff
@@ -221,17 +221,17 @@ class Plugin {
   virtual void AdvanceTime(Instant const& t, Angle const& planetarium_rotation);
 
   // Returns the degrees of freedom of the given part in |World|, assuming that
-  // the origin of |World| is fixed at |bubble_barycentre_|.  The part must be
-  // in a loaded vessel.
```
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Was it intentional to remove the last sentence?

Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: It was, this invariant is not enforced, and has no fundamental reason to be. Eventually we will distinguish loaded and packed and we can think about that again.

**Pull request merged**